pkg/asset/installconfig/gcp/dns.go

```diff
@@ -0,0 +1,28 @@
+package gcp
+
+import (
+	"golang.org/x/net/context"
+	"google.golang.org/api/dns/v1"
+)
+
+// GetPublicZone returns a DNS managed zone from the provided project which matches the baseDomain
+func GetPublicZone(baseDomain, project string) (managedZone *dns.ManagedZone, err error) {
+	ctx := context.Background()
+	//TODO: Determine if this will be proper/acceptable way to invoke services with credentials
+	dnsService, err := dns.NewService(ctx)
+	var res dns.ManagedZone
+	req := dnsService.ManagedZones.List(project)
+
+	//TODO: How are we selecting managed zones? This returns the last match.
+	// Check for public/private?
```

Comment:
https://github.com/openshift/installer/blob/45f81e1d950cbc80e46e0b230e11b0032de8c868/pkg/asset/installconfig/aws/basedomain.go#L82

It needs to be public.